bitcoin/src/address.rs

```diff
@@ -643,9 +505,193 @@ impl NetworkValidation for NetworkUnchecked {
 /// representation of an address without the network validation tag, which is used to ensure that
 /// addresses are used only on the appropriate network.
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-struct AddressInner {
-    payload: Payload,
-    network: Network,
+enum AddressInner {
+    P2pkh {
+        hash: PubkeyHash,
+        prefix: LegacyP2pkhPrefix,
+    },
+    P2sh {
+        hash: ScriptHash,
+        prefix: LegacyP2shPrefix,
```

Comment:
Since they are already solidly considered legacy (are we really going to create more of them?), aren't `LegacyP2pkhPrefix` and `LegacyP2shPrefix` always going to be the same? Is the issue to find a name for them? I agree `[Legacy]Base58Prefix` isn't ideal. `PreSegwitPrefix`?

Reply:
Seems they are different:
```rust
/// Mainnet (bitcoin) pubkey address prefix.
pub const PUBKEY_ADDRESS_PREFIX_MAIN: u8 = 0; // 0x00
/// Mainnet (bitcoin) script address prefix.
pub const SCRIPT_ADDRESS_PREFIX_MAIN: u8 = 5; // 0x05
/// Test (tesnet, signet, regtest) pubkey address prefix.
pub const PUBKEY_ADDRESS_PREFIX_TEST: u8 = 111; // 0x6f
/// Test (tesnet, signet, regtest) script address prefix.
pub const SCRIPT_ADDRESS_PREFIX_TEST: u8 = 196; // 0xc4
```